### DISCUSSION TITLE
How does hyperdrive actually work? (A proposal/discussion.)

### DISCUSSION THREAD

**Discussion by MasterOfGrey** [endless-sky/endless-sky#7077] [Lore]
This branches a little bit from the conversation in #7061 and from a theory I was working with for an outfit in #6439 - a thought to establish some baseline assumptions for what a hyperlane actually is, at least enough to stick to some consistent interpretations that can be used in writing.

"Hyperspace" is a common enough trope in sci-fi, and it's usually something you _enter_ - some other plane of reality.

However, we have this from MZ's lore document:

> Spacetime is composed of two layers which the Quarg refer to as the “warp” and the “weft.” (This is a textile analogy.)
> The warp is the ordinary, linear spacetime that human beings, etc. inhabit. Travel within the warp is limited to the speed of light, but technology can be used to build paths between any two sufficiently massive objects (e.g. stars) where the distance is artificially condensed. These paths are what humans refer to as “hyperspace lanes,” although most humans believe them to be natural phenomena. Building these links requires matched quantum-linked transmitters, so you can’t build a link to a new star system without first sending a transmitter there by other means (either slower-than-light travel or jump drive).
> The weft, on the other hand, is a dimension where space and time are nonlinear. By briefly passing through weft-space, a ship can instantaneously travel a significant distance in linear space. The jump drive is the technology that allows this sort of travel. It relies on exotic technology that few species can replicate and fewer still understand. For example, the Korath exiles (tier 2.5) are able to build jump drives in limited numbers after reverse-engineering Quarg technology, but they suffer frequent destructive manufacturing mishaps.
> Hyperspace (warp) travel, on the other hand, is accessible to a tier 1 civilization, but the technology to create hyperspace lanes is much more advanced, roughly tier 3; it’s more or less a matter of luck whether a species discovers it before discovering the jump drive. The Quarg, for example, never discovered how to build hyperspace lanes. (But, the Archons are capable of constructing them.)
> For a very long time, advanced life did not evolve in most star systems because the early planetary systems did not contain enough of the heavier elements. It was only after several generations of stars began to go nova, seeding the galaxy with the elements produced in their cores through nuclear fusion, that life even became possible...

^All of that implies that "hyperspace" isn't actually separate from our own reality at all, it's simply a modified section of real space that defies explanations from classical or maybe even quantum physics - operating in a higher band of math than we've yet developed/encountered.

This would suggest that you don't actually _enter_ hyperspace. Visually in-game, it certainly doesn't appear like you do any such thing. With the way things are now it feels more like you just "surf" on regular space using some kind of hyperplane that's built into this reality - which also harks back to the differences between the warp and weft and is consistent with the explanations from the above quoted section of the lore document.

I would like to suggest that a hyperlane, is therefore "just" some kind of gravitational slipstream modifying the behaviour of real space. Further, continuing the textile analogy, I would like to suggest that it forms a kind of self-enclosed tunnel which is consistent - a cord of modified space looping in on itself from A to B which essentially "frays" at the ends. 

This fraying would explain why you have to stabilise enough to lock onto it with a hyperdrive, and then expend power to go into it instead of simply "falling into" a gravity slipstream. <- The slipstream would be inaccessible except by following a frayed thread right into the end of the lane. (And very slight misalignments of that would result in annihilation, which would also explain early hyperdive accidents.)

One interesting corollary of this is that your ship wouldn't be visible from outside the system as it travelled along the lane, but from the system at either end it _would_ be visible once/while its relative speed was < light speed as it accelerated out/decelerated in.
(Which would also be consistent with existing in-game experiences.)

Thoughts?

**Comment by Zitchas** [comment]
It might be a combination, actually. Hyperspace, or weft space as described here, could very well be a separate space that is strongly connected to the "warp." To borrow alternate terminology prevalent in the fantasy genre (specifically, Forgotten Realms, and more generally throughout the Dungeons and Dragons and Pathfinder systems) there is the Prime Material plane (aka the reality we all deal with), and the Ethereal Plane. Much of the Ethereal plane coincides with the physical, although distance isn't as much of a meaningful concept in it. That being said, it is almost more of a scale of how far one is into it. Lots of spells, for instance, allow people to see into or move into the Ethereal plane, where they are basically still in the Prime Material but become ghostlike and insubstantial. Moving farther into it and they lose the ability to interact with the Prime Material, but gain the ability to cross vast distances in short periods of time. Move far enough through this border into the Ethereal plane proper, and distance - as far as things on the Prime Material are concerned - becomes virtually meaningless.

Drawing on this idea, I see the weft (hyperspace) as being its own space apart from the Prime Material, but heavily interconnected from it. The closer one is to super-massive gravitational nodes (aka stars, black holes, and other stellar objects), the more "aligned" it becomes with the warp (prime material). Rather like weaving, where the farther a thread gets from an anchor, the more free-form it can be.

Tying this in with your suggestion, then hyperlanes could originally be dating back to the big bang and the primordial soup, where the universe was much more closely packed together. As it separated, adjacent concentrations of matter could form these connections with each in a similar manner to how one could use quantum linked transmitters. Perhaps the natural formation of these lanes through the expansion of the universe only form between not between stars, but rather the combine masses of entire star clusters, perhaps even just the largest of star clusters. As such, our current star map with its hyperlinks that span hundreds or even thousands of light years bypassing the vast majority of stars is because the links aren't going from star to star, they are going from center of cluster to center of cluster. The exact location of the end might have gravitated to being directly on whichever star is closest to the center of that cluster, giving the illusion that it is going from star to star, but needed actually be the case. And the gravitational effects of dark matter and other exotic stuff would likewise affect the formation and placement of these connections.

Beyond that, hyperlanes, as you describe, are slipstreams or tunnels of "warp" space that cross over into the "weft," which brings with it many of the attributes of regular warp space. The result is these tunnels of space that are hybrids of warp and weft that follow many (maybe even most) of the rules of warp space, but with a meshing of exotic effects overtop of it, allowing the crossing of vast distances in moments.

Thus hyperdrives (very much like the slipstream drives from Andromeda) are multi-function devices that provide hyperlink locating, hyperlane entry navigation/calculation, and the mechanism for triggering entry into it.

Jump Drives, on the other hand, are devices that ignore the hyperlanes themselves to locate nodes of warp/weft interface, from some sort of "bubble" of warp space around the ship, and catapult it through the weave to be "caught" by the node at the other end.

And then there is the Drak mythology (aka stuff that isn't in-game but is in lore documents and such) that suggest that it is possible to simply enter the weft outright. Not as a bubble of warp space being flung from one node to another, but just straight up transferring something from warp space into weft space indefinitely. (this is not a drive we see in use anywhere in-game, with the possible exception of the Archon's death-escape manoeuvre/animation.

**Comment by MasterOfGrey** [comment]
> Hyperspace, or weft space as described here

Jump space is weft space. Hyperspace was explicitly an alteration of the warp.

**Comment by roadrunner56** [comment]
PR #6761 may have some relevance to the discussion here, as it references the weft and hyperspace.

**Comment by Chopliver19** [comment]
Kinda like in Minecraft where one block in the Nether equals 8 blocks in the overworld?

**Reply by mOctave** [reply]
I think the hyperdrive, at least, works more like the Wrinkle in Time tesseract: it allows you to move from point A to point B by pinching the eight-dimensional cloth made by warp and weft space in a ninth dimension so that you can travel through that ninth dimension instead of having to use the other eight to navigate.

The jump drive, on the other hand, would allow you to travel through the four dimensions of weft space to let you back into warp space, although I'm not sure whether this is because of distances being changed or distances being nonexistent.

**Comment by ChuckStarchaser** [comment]
I'm not a connoisseur of the lore in this matter;  but I think I can intuit a few things regarding the _original idea_ of hyper-space as well as sub-space.
Consider that these terms go back quite some time, and probably predate the time when worm-holes became popular.

I would think _hyper-space_ meant "above (normal) space", just as _sub-space_ meant "below (normal) space",  where 'above' and 'below' are in reference to a fourth dimension.  This is not so much gobbledygook as stuff perfectly compatible with General Relativity thinking and its implications.

In General Relativity, "Space" is not non-Cartesian simply because the Cartesian axes get warped somehow.  The Cartesian axes are actually quite straight.  What bends is AN OBJECT that we call "space".  This OBJECT is non-flat.
Most people, even physicists, don't get the idea of what Einstein was talking about when he introduced the concept of space not being "flat".  He tried to explain it with an example about Flatlanders that live in a 2D world, and that after one of them discovers that while the inner angles of all triangles add to 180 degrees "most of the time", if you measure the angles of very large triangles, the sum begins to creep above 180 degrees.  And he comes up with the theory that their 2D space is non-flat.
Now, most Flatlanders mis-interpret the theory, and think that 2D Space, itself, somehow warps.  He tries to tell them "No, no, no;  it is not that 2D space is warped;  it's just that we live on the surface of a 3D object, possibly a sphere.
  Of course they don't know the nature of this object.  We, 3D beings, could tell them "you are living on a big balloon made of rubber", but it would mean little to a 2D being that never saw a 2D version of rubber.

Similarly, what Einstein meant is that we live on the 3D surface of a 4D object of some form.  We confusingly call it "space", but it is not an abstraction like "Cartesian Space";  it is an object.  This 4D object could be solid or liquid.  Probably liquid.  The waves on this 4D liquid propagate through the 3D surface spherically, and possibly those could be our electromagnetic waves, or gravity waves.

 Now, just as we have hover-craft that can go faster than surface ships, as they don't cause too much waves, and therefore waste less energy,  so we could conceivably have a way for a 3D space-ship to rise in the 4th dimension, above the water (above "normal space), and so be able to move faster than the speed of light, since the speed of light is literally the speed of light waves, which occur on the surface.  The space above the surface does not, presumably, have a dielectric constant and magnetic permeability, like regular space does, which is what determines the speed of light;  and so the speed of light limit does not apply to hyper-space.

And now, just as we have submarines that can also go faster than surface ships, since they also avoid wasting energy in producing surface waves, so we could also conceivably have 3D space-ships that can somehow dive below "normal space", where, again, the speed of light limit doesn't apply.  This is sub-space.

There was an article on Scientific American, a few years ago, about observations of space near the event horizons of black holes, that seem to point out that space is not a continuum, but that it rather has a "grainy" structure.  What I think that means is that the above concept is being slowly discovered to be true.

I've always had the feeling that THIS is what the original hyper-space and sub-space concepts were.  How they have evolved (or devolved) in time and lore is another story.

As for hyper-space LANES ...  Oh my!  Well, that came about to make games easier to code, and fiction easier to write.  If a former civilization, very advanced, built some kind of network, and we just discovered it, it saves us from having to explain how it works;  doesn't it?  It's an old trick...


How to use these concepts in fiction:

The difference between sub-space and hyper-space is that sub-space exists within the denser 4D medium, while hyper-space exists within the lighter 4D medium.  It's as if normal space is the surface of the ocean, hyper-space is the air above, and sub-space is inside the water below.  Thus, we could theorize that if there are any frictions or speed limits in these two off-normal-space regions, the limits would be lower for sub-space.  And so, perhaps sub-space would only allow you to move at 5 times the speed of light, whereas hyper-space might be almost unlimited in speed.  But perhaps sub-space is easier to get into than hyperspace.

The way I would use these concepts, in some game, or book or movie, would be to make sub-space the standard way of moving intra-system, while making hyper-space the standard for inter-system, and would give them different sets of rules, and technologies.

Of course, the elephant in the room is how do we accelerate the mass of the ship ...  Even with fusion power and whatnot, it would take a long time accelerating before you reach the speed of light, let alone warp 3 ...
